kirby/api/ext.py

```diff
@@ -198,32 +198,24 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self._session.close()
 
-    @tenacity.retry(**webserver_retry_args)
-    def post(self, endpoint, data, params=None):
-        if not params:
-            params = {}
-        result = self._session.post(
-            urljoin(self.web_endpoint_base, endpoint), data=data, params=params
-        )
-        if result.status_code == 200:
-            return result.json()
-        raise WebClientError(
-            f"POST error on {result.url}. "
-            f"Status code : {result.status_code}. "
-            f"Response : {result.text}"
-        )
+    def _request_decorator(self, method):
+        @tenacity.retry(**webserver_retry_args)
+        def _request(endpoint, **kwargs):
+            result = method(
+                urljoin(self.web_endpoint_base, endpoint), **kwargs
+            )
 
-    @tenacity.retry(**webserver_retry_args)
-    def get(self, endpoint, params=None):
-        result = self._session.get(
-            urljoin(self.web_endpoint_base, endpoint), params=params
-        )
-        if result.status_code == 200:
-            json = result.json()
-            if json:
-                return json
-        raise WebClientError(
-            f"GET error on {result.url}. "
-            f"Status code : {result.status_code}. "
-            f"Response : {result.text}"
-        )
+            if result.status_code == 200:
+                return result.json()
+            raise WebClientError(
+                f"{method} error on {result.url}. "
+                f"Status code : {result.status_code}. "
+                f"Response : {result.text}"
+            )
+
+        return _request
+
+    def __getattr__(self, item):
```

Comment:
Can you a test where you test the most common verbs (e.g. get, post, put, delete, head, options) please? Just to make sure they work (but obviously you can mock the actual calls)

Reply:
Test written. Also, now testing errors handling with the same principle